File: MojPrijevoz/MojPrijevoz.Model/Requests/UserVehicle/UserVehicleUpsertRequest.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace MojPrijevoz.Model.Requests.UserVehicle;

public class UserVehicleUpsertRequest
{
    [Required] public int VehicleId { get; set; }

    [Required] public int ModelYear { get; set; }

    [MaxLength(9)] [Required] public string LicensePlate { get; set; } = null!;

    [Required] [Range(0, 100)] public float PricePerKm { get; set; }

    [JsonIgnore] public int? ProfileId { get; set; }
    [JsonIgnore] public bool IsFirstVehicle { get; set; } = false;
}

File: MojPrijevozFront/moj_prijevoz/lib/resources/requests/user_vehicle/user_vehicle_upsert_request.dart
import 'package:dio/dio.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:moj_prijevoz/resources/requests/has_picture_interface.dart';
import 'package:moj_prijevoz/utils/json_parser.dart';

part 'user_vehicle_upsert_request.g.dart';

@JsonSerializable()
class UserVehicleUpsertRequest extends JsonRequest
    implements HasPictureInterface {
  int? vehicleId;
  int? modelYear;
  String? licensePlate;
  double? pricePerKm;

  @override
  @JsonKey(includeFromJson: false, includeToJson: false)
  MultipartFile? picture;

  UserVehicleUpsertRequest({
    this.vehicleId,
    this.modelYear,
    this.licensePlate,
    this.pricePerKm,
    this.picture,
  });

  @override
  Map<String, dynamic> toJson() => _$UserVehicleUpsertRequestToJson(this);

  factory UserVehicleUpsertRequest.fromJson(Map<String, dynamic> json) =>
      _$UserVehicleUpsertRequestFromJson(json);
}
